flycheck.el

```diff
@@ -6068,6 +6069,27 @@ See URL `http://www.foodcritic.io'."
        ;; http://matschaffer.github.io/knife-solo/#label-Init+command
        (locate-dominating-file parent-dir "cookbooks")))))
 
+(flycheck-define-checker chicken-scheme
+  "A CHICKEN Scheme syntax checker using CHICKEN compiler `csc'.
+
+See URL `http://call-cc.org/'."
+  :command ("csc" "-A" "-local" source-inplace)
+  :predicate
+  (lambda ()
+    (or (not (eq major-mode 'scheme-mode))
+        ;; In `scheme-mode' we must check the current Scheme implementation
+        ;; being used
+        (and (boundp 'geiser-impl--implementation)
+             (eq geiser-impl--implementation 'chicken))))
+  :error-patterns
+  ((info line-start "Note: " (zero-or-more not-newline) ":\n  "
+         "(" (file-name) ":" line ") " (message) line-end)
+   (warning line-start "Warning: " (zero-or-more not-newline) ":\n  "
+            "(" (file-name) ":" line ") " (message) line-end)
+   (error line-start "Error: " (zero-or-more not-newline) ":\n  "
+          "(" (file-name) ":" line ") " (message) line-end))
```

Comment:
In all of these patterns please don't hard-code the expected indentation after the line break, but use `(one-or-more (any space))` instead.  That's more tolerant of whitespace changes in the output.